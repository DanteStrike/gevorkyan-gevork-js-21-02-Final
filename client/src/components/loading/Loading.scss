.loading {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 100;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto;
  color: var(--primary-font-color);
  background-color: var(--primary-background);
  border-radius: inherit;
}
