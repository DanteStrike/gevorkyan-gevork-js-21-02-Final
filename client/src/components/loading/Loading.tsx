import React from 'react';
import {Spin} from 'antd';
import './Loading.scss';

interface ILoadingProps {
  isLoading: boolean;
}

function Loading({isLoading}: ILoadingProps) {
  return (
    <>
      {isLoading && (
        <div className="loading">
          <Spin size="large" />
        </div>
      )}
    </>
  );
}

export default React.memo(Loading);
