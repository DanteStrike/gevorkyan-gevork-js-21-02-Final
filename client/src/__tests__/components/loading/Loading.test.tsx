import {shallow} from 'enzyme';
import React from 'react';
import Loading from '../../../components/loading/Loading';

describe(`Component Loading should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Loading isLoading={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot loading test`, () => {
    const wrapper = shallow(<Loading isLoading />);
    expect(wrapper).toMatchSnapshot();
  });
});
