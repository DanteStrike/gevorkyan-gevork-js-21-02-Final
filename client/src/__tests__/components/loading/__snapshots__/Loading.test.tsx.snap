// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loading should render correctly snapshot default test 1`] = `<Fragment />`;

exports[`Component Loading should render correctly snapshot loading test 1`] = `
<Fragment>
  <div
    className="loading"
  >
    <Spin
      size="large"
      spinning={true}
      wrapperClassName=""
    />
  </div>
</Fragment>
`;
